### PR TITLE
only patch authenticators when there's no webhook authenticators

### PR DESCRIPTION
**Pull request by stlaz**
As designed in: https://github.com/openshift/enhancements/pull/142

### CODE REVIEW

File: vendor/k8s.io/kubernetes/pkg/kubeapiserver/authenticator/config.go [hunk 1]
```diff
@@ -189,14 +189,15 @@ func (config Config) New() (authenticator.Request, *spec.SecurityDefinitions, er
 		}
 
 		tokenAuthenticators = append(tokenAuthenticators, webhookTokenAuth)
+	} else {
+		// openshift gets injected here as a separate token authenticator because we also add a special group to our oauth authorized
+		// tokens that allows us to recognize human users differently than machine users.  The openAPI is always correct because we always
+		// configuration service account tokens, so we just have to create and add another authenticator.
+		// TODO make this a webhook authenticator and remove this patch.
```
Comment (mfojtik) [inline vendor/k8s.io/kubernetes/pkg/kubeapiserver/authenticator/config.go:196]: please open a BZ

Comment (stlaz) [reply vendor/k8s.io/kubernetes/pkg/kubeapiserver/authenticator/config.go:196]: Added enhancement PR link instead

**Comment by marun** [comment]
/lgtm

@stlaz Can I squash this commit with the one introducing the patch during the rebase?

**Comment by stlaz** [comment]
> Can I squash this commit with the one introducing the patch during the rebase?

sure, I would even advise to do that :)

File: vendor/k8s.io/kubernetes/pkg/kubeapiserver/authenticator/config.go (hunk 1 above)
Comment (sttts) [reply vendor/k8s.io/kubernetes/pkg/kubeapiserver/authenticator/config.go:196]: This is part of our move to webhooks in general as described in the enhancement. No need to track that in BZ.

**Pull request merged**